eth_utils/address.py

```diff
@@ -188,18 +177,25 @@ def to_checksum_address(address):
         )
         for i in range(2, 42)
     ))
-    return checksum_address
+    if is_bytes(checksum_address):
```

Comment:
`add_0x_prefix` should never be returning `bytes`. Why can't we just return `checksum_address` here?